### PR TITLE
KIALI-2306 Add detailed traffic view to Applications page

### PR DESCRIPTION
**Pull request by israel-hdez**
https://issues.jboss.org/browse/KIALI-2306

**This is a follow-up of https://github.com/kiali/kiali-ui/pull/956** That PR goes first. ~~This one will also show changes for that PR.~~

image

### CODE REVIEW

File: src/pages/AppDetails/AppDetailsPage.tsx [hunk 1]
```diff
@@ -69,16 +89,48 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
...
+      namespace: { name: this.props.match.params.namespace },
+      nodeType: NodeType.APP,
+
+      // unneeded
+      workload: '',
+      service: '',
+      version: ''
+    };
+    const restParams = {
+      duration: '600s',
```
Comment (jshaughn) [inline src/pages/AppDetails/AppDetailsPage.tsx]: "duration" is the correct term for this, but it is confusingly referred to as "rateInterval" below, in TrafficDetails.   I think you should rename the "rateInterval" property in TrafficDetails to "duration".  Also, I'd suggest not using the 600 constant in multiple places.  Perhaps define a single constant and use it here, and also below.

Note that "rateInterval" should be used when working with the actual rateInterval to be applied in a rate() calculation with Prometheus.

Although it's not a blocker, I wonder if this fetch of data should be done in the TrafficDetails component when the tab is selected, and not proactively here.

These comments also apply to the former Workload details.

File: src/pages/AppDetails/AppDetailsPage.tsx [hunk 2]
```diff
@@ -100,12 +152,22 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
...
-                  onRefresh={this.fetchApp}
+                  onRefresh={this.doRefresh}
                   activeTab={this.activeTab}
                   onSelectTab={this.tabSelectHandler}
                   health={this.state.health}
                 />
               </TabPane>
+              <TabPane eventKey="traffic">
+                <TrafficDetails
+                  rateInterval={600}
```
Comment (jshaughn) [inline src/pages/AppDetails/AppDetailsPage.tsx]: See the comment above about renaming this property and replacing this 600 with a constant.  I know that we'll probably be changing this to use a duration dropdown but until that time, a constant would be better to ensure the graph query and the displayed fetch duration are the same.

Comment (israel-hdez) [reply src/pages/AppDetails/AppDetailsPage.tsx]: Ok, I fixed the naming.

File: src/pages/AppDetails/AppDetailsPage.tsx (hunk 1 above)
Comment (israel-hdez) [reply src/pages/AppDetails/AppDetailsPage.tsx]: You are right about the constants. I fixed that.

And yes... Initially I wanted to keep the `TrafficDetails` component stateless. But it does have some state. So, I think it's completely valid to move the API calls to `TrafficDetails`. I'll add that to my TODO list.

**Comment by israel-hdez** [comment]
@jshaughn I fixed your comments. Regarding the workload-to-workload comment, I'm sure it won't work. It won't crash, but there will be no row showing that traffic :( I wonder if you prefer the fix to be on this PR or to open another one.

**Comment by israel-hdez** [comment]
@jshaughn I also rebased to get the last design of the traffic component.

File: src/components/Metrics/TrafficDetails.tsx [hunk 3]
```diff
@@ -58,7 +60,7 @@ class TrafficDetails extends React.Component<TrafficDetailsProps, TrafficDetails
   }
 
   render() {
-    const rateIntervalName = getName(this.props.rateInterval).toLowerCase();
+    const rateIntervalName = getName(this.props.duration).toLowerCase();
```
Comment (jshaughn) [inline src/components/Metrics/TrafficDetails.tsx]: To be consistent, I think this should be durationName or maybe durationLabel.

Comment (israel-hdez) [reply src/components/Metrics/TrafficDetails.tsx]: Ouch. I missed that one. Fixed!

**Comment by israel-hdez** [comment]
@hhovsepy Do you have the time to check this one? It's quite similar to the traffic workloads PR.

**Comment by israel-hdez** [comment]
Merging. Thanks for your reviews @jshaughn @hhovsepy .

**Pull request merged**